dictionaryutils/__init__.py

```diff
@@ -0,0 +1,185 @@
+from copy import deepcopy
+from collections import namedtuple
+from contextlib import contextmanager
+from jsonschema import RefResolver
+
+import glob
+import logging
+import requests
+import os
+import yaml
+
+MOD_DIR = os.path.abspath(os.path.dirname(__file__))
+
+ResolverPair = namedtuple('ResolverPair', ['resolver', 'source'])
+
+
+@contextmanager
+def visit_directory(path):
+    """Perform contained actions with current working directory at
+    :param:``path``.  Always return to previous directory when done.
+
+    """
+
+    cdir = os.getcwd()
+    try:
+        os.chdir(path)
+        yield os.getcwd()
+    finally:
+        os.chdir(cdir)
+
+
+def load_yaml(name):
+    """Return contents of yaml file as dict"""
+    with open(name, 'r') as f:
+        return yaml.safe_load(f)
+
+
+def load_schemas_from_url(url, logger):
+    try:
+        r = requests.get(url)
+    except:
+        logger.exception('Fail to get schema from {}'.format(url))
+        raise
+    if r.status_code != 200:
+        logger.error('Fail to get schema from {}: {}'.format(url, r.text))
+        raise
+    schemas, resolvers = {}, {}
+
+    for key, schema in r.json().iteritems():
+        schemas[key] = schema
+        resolver = RefResolver('{}#'.format(key), schema)
+        resolvers[key] = ResolverPair(resolver, schema)
+
+    return schemas, resolvers
+
+
+def dump_schemas_from_dir(directory):
+    """Dump schema as a json"""
+
+    schemas = {}
+    with visit_directory(directory):
```

Comment:
can write as:
```python
with visit_directory(directory):
    return {path: load_yaml(path) for path in glob.glob('*.yaml')}
```
(also applies in `load_schemas_from_dir` below)
Do we need to handle case if any files are `"*.yml"`?

Reply:
nah, no reason to use two formats, but :+1: for the first suggestion